runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -768,10 +812,38 @@ private void addCounter()
 			taskTooltip += taskLocation + "</br>";
 		}
 
-		taskTooltip += ColorUtil.wrapWithColorTag("Pts:", Color.YELLOW)
-			+ " %s</br>"
-			+ ColorUtil.wrapWithColorTag("Streak:", Color.YELLOW)
-			+ " %s";
+		String slayerItemName = "";
+		int slayerItemID = Task.getTask(taskName).getWeaknessItem();
+		Color itemColor;
+
+		if (slayerItemID != -1)
+		{
+			if (itemOnPlayer(slayerItemID))
+			{
+				itemColor = Color.GREEN;
+			}
+			else
+			{
+				itemColor = Color.RED;
+			}
+
+			slayerItemName = client.getItemDefinition(Task.getTask(taskName).getWeaknessItem()).getName();
+
+			taskTooltip += ColorUtil.wrapWithColorTag("Slayer Item: ", Color.YELLOW)
+				+ ColorUtil.wrapWithColorTag(slayerItemName, itemColor)
+				+ " </br>"
+				+ ColorUtil.wrapWithColorTag("Pts:", Color.YELLOW)
+				+ " %s</br>"
+				+ ColorUtil.wrapWithColorTag("Streak:", Color.YELLOW)
+				+ " %s";
```

Comment:
This piece of the tooltip is identical in both branches; remove the `else` branch and append this part of the string after the slayerItemID check.